src/v/kafka/client/types.h

```diff
@@ -30,4 +32,42 @@ inline constexpr model::node_id consumer_replica_id{-1};
 /// \brief during connection, the node_id isn't known.
 inline constexpr model::node_id unknown_node_id{-1};
 
+/// \brief Structure used to report metadata updates
+/// Both describe cluster and metadata use very similar fields but are typed
+/// differently
+struct metadata_update {
```

Comment:
This type doesn't differentiate between when topics were not requested vs they are not present